torch/csrc/jit/codegen/cuda/tensor_view.cpp

```diff
@@ -680,46 +641,41 @@ TensorView* TensorView::cache_before() {
   }
 
   // Create Producer Domain
-  // This domain will be the consumer, so create the producer
+  // This domain will be the consumer which needs a new domain, so replace the
+  // producers domain with this domain.
   auto root_domain = getRootDomain();
-  TensorView* producer = new TensorView(
-      new TensorDomain(
-          IterDomain::clone(root_domain),
-          std::vector<bool>(root_domain.size(), true)),
-      getDataType().value());
+
+  TensorView* producer = nullptr;
+  if (hasRFactor()) {
```

Comment:
I'm not sure how this works when the consumer has RFactor. The consumer loses the RFactor root domain, so that seems to cause root mapping to fail between the consumer and its consumer.

Reply:
You're right, I just mechanically copied the constructors. Above asserts prevent this from ever acting on an rfactor domain. Will remove the rfactor constructor.